game/src/main/kotlin/world/gregs/voidps/world/activity/dnd/shootingstar/ShootingStar.kts

```diff
@@ -0,0 +1,332 @@
+import world.gregs.voidps.engine.entity.worldSpawn
+import world.gregs.voidps.engine.inject
+import world.gregs.voidps.engine.inv.add
+import world.gregs.voidps.engine.inv.holdsItem
+import world.gregs.voidps.engine.inv.inventory
+import world.gregs.voidps.engine.inv.remove
+import world.gregs.voidps.engine.queue.softQueue
+import world.gregs.voidps.engine.suspend.pause
+import world.gregs.voidps.network.visual.update.player.EquipSlot
+import world.gregs.voidps.type.Direction
+import world.gregs.voidps.type.Tile
+import world.gregs.voidps.type.random
+import world.gregs.voidps.world.activity.bank.bank
+import world.gregs.voidps.world.interact.dialogue.Cheerful
+import world.gregs.voidps.world.interact.dialogue.Sad
+import world.gregs.voidps.world.interact.dialogue.type.npc
+import kotlin.math.roundToInt
+import kotlin.random.Random
+
+val itemDefinitions: ItemDefinitions by inject()
+val objects: GameObjects by inject()
+val npcs: NPCs by inject()
+
+var totalCollected: Int = 0
+var currentStarTile = Tile(-1, -1) // same as below I don't know if this is a good way to init this var
+var currentActiveObject = GameObject(-1,-1,-1,-1,-1,-1) // I don't know if this is a good way to init this var
+val startEvent = Random.nextInt(from = 1 * 60 * 60 * 1000, until = 2 * 60 * 60 * 1000) // get a random time between 1h - 2h
+val initialStar: IntArray = intArrayOf(38660, 38661, 38662, 38663, 38664, 38665, 38666, 38667, 38668) // array of all tiers of stars to be picked randomly
+val ownedStarDustThreshold = 200 // the max amount of star dust a player is allowed to own in both inventory and bank
+
+worldSpawn {
+    eventUpdate()
+}
+
+fun eventUpdate() {
+    World.queue("shooting_star_event_timer", startEvent) {
+       if(isPlayersPresent()) {
+            eventUpdate()
+            return@queue
+       }
+       if(currentStarTile != Tile(-1, -1)) { //there's already an active event
+            cleanseEvent(true) // force stop even meaning - don't spawn the npc because the server stopped the event.
+            println("There was already an active star, deleted it and started a new event")
+       }
+       startCrashedStarEvent()
+       eventUpdate()
+    }
+
+}
+
+fun isPlayersPresent(): Boolean {
+    return false
+}
+
+fun startCrashedStarEvent() {
+    val starLocation = getRandomLocation()
+    val starToSpawn = getInitialStar()
+    currentStarTile = starLocation.location
+    currentActiveObject = GameObject(starToSpawn, currentStarTile, 10, 0)
+    objects.add(currentActiveObject)
+    println("${starLocation.locationDescription}")
+}
+
+fun handleMinedStarDust(currentMinedStar: GameObject) {
+    val starPayout = currentMinedStar.def["collect_for_next_layer", -1]
+    if(totalCollected >= starPayout){
+        val stage = currentMinedStar.id.takeLast(1)
+        if(stage.equals("1")){ //if is the lowest tier star just delete the star and clean the event
+            currentMinedStar.remove(currentMinedStar.intId, true)
```

Comment:
```suggestion
            currentMinedStar.remove()
```
The first parameter in [`GameObject.remove()`](https://github.com/GregHib/void/blob/d491ca6b2779b47fe7b603776d0cea32b073b7ef/engine/src/main/kotlin/world/gregs/voidps/engine/entity/obj/GameObjects.kt#L375) is number of ticks to remove the object for, -1/empty to remove forever